packages/imperative/src/config/cmd/report-env/EnvQuery.ts

```diff
@@ -0,0 +1,624 @@
+import { CliUtils } from "../../../../../utilities/src/CliUtils";
+
+import { IPluginJson } from "../../../plugins/doc/IPluginJson";
+import { PluginIssues } from "../../../plugins/utilities/PluginIssues";
+
+import { ItemId, IProbTest, probTests } from "./EnvItems";
+
+/**
+ * This interface represents the result from getEnvItemVal().
+ */
+export interface IGetItemVal {
+    itemVal: string;        // Value of the item. Null when we cannot get the value.
+    itemValMsg: string;     // Message to display the item's value.
+    itemProbMsg: string;    /* Message to display any problems with the value.
+                             * Empty string (length 0) when there are no problems.
+                             */
+}
+
+/**
+ * This interface represents the the options for getEnvItemVal().
+ */
+export interface IGetItemOpts {
+    /**
+     * API to start/end progress bar for long running actions. Since this is a CLI thing,
+     * it is optional in case non-CLI programs want to call the getEnvItemVal function.
+     */
+    progressApi?: IHandlerProgressApi;   // API to start/end progress bar
+}
+
+/**
+ * This class encapulates operations for Zowe CLI environment information.
+ * We use the term environment loosely. Sometimes it is an environment variable.
+ * It can also be something in the runtime environment, like version of NodeJS.
+ */
+export class EnvQuery {
+    private static readonly divider = `______________________________________________${os.EOL}`;
+    private static readonly indent = "    ";
+    private static readonly eolMatch: string = "(\r?\n|\n)";
+    private static readonly allEolRegex = new RegExp(EnvQuery.eolMatch, "g");
+    private static readonly lastEolRegex = new RegExp(EnvQuery.eolMatch + "$");
+
+    // __________________________________________________________________________
+    /**
+     * For the specified itemId, get its value.
+     *
+     * @param itemId ID of the environmental item for which we want get the value.
+     * @param getItemOpts options that affect our behavior.
+     *
+     * @returns An object with the item value, a display message, and a problem message.
+     */
+    public static async getEnvItemVal(itemId: ItemId, getItemOpts: IGetItemOpts = {}): Promise<IGetItemVal> {
+        const getResult: IGetItemVal = { itemVal: null, itemValMsg: "", itemProbMsg: "" };
+        switch(itemId) {
+            case ItemId.ZOWE_VER: {
+                EnvQuery.getZoweVer(getResult);
+                break;
+            }
+            case ItemId.NODEJS_VER: {
+                getResult.itemVal = process.versions.node;
+                getResult.itemValMsg = "Node.js version = " + getResult.itemVal;
+                break;
+            }
+            case ItemId.NVM_VER: {
+                getResult.itemVal = EnvQuery.getCmdOutput("nvm", ["version"]);
+                getResult.itemValMsg = "Node Version Manager version = " + getResult.itemVal;
```

Comment:
Please note that Node.js does not process aliases when spawning/executing a command.
In this particular scenario, It says that I don't have `nvm` even though it's in `.bashrc` 😋 
I don't know of a good way around this though 😅